orttraining/orttraining/core/graph/optimizer/lamb_optimizer_builder.cc

```diff
@@ -9,13 +9,14 @@
 #include "core/framework/tensorprotoutils.h"
 #include "core/util/math.h"
 #include "onnx/defs/attr_proto_util.h"
+#include "orttraining/core/session/training_session.h"
 
 namespace onnxruntime {
 namespace training {
 Status LambOptimizerBuilder::Build(
     const OptimizerBuilderConfig& config,
     GraphAugmenter::GraphDefs& graph_defs,
-    std::vector<ONNX_NAMESPACE::TensorProto>& new_external_initializers,
+    std::unordered_map<std::string, std::vector<ONNX_NAMESPACE::TensorProto>>& weight_to_opt_mapping,
```

Comment:
Why is ```weight_to_opt_mapping``` being represented as a ```std::unordered_map<std::string, std::vector<ONNX_NAMESPACE::TensorProto>>```? Can we ensure uniqueness of the ```TensorProto``` in the vector?

Reply:
Because this value doubles as the mapping info of weight to optimizers and the actual tensor value of optimizers. The values are separated in orttraining/orttraining/core/graph/optimizer_graph_builder.cc.
`TensorProto` is unique object with different optimizer names